packages/@magic-sdk/provider/src/util/device-share-web-crypto.ts

```diff
@@ -0,0 +1,120 @@
+import { getItem, iterate, removeItem, setItem } from './storage';
+import { isWebCryptoSupported } from './web-crypto';
+
+export const DEVICE_SHARE_KEY = 'ds';
+export const ENCRYPTION_KEY_KEY = 'ek';
+export const INITIALIZATION_VECTOR_KEY = 'iv';
+
+const ALGO_NAME = 'AES-GCM'; // for encryption
+const ALGO_LENGTH = 256;
+
+export async function clearDeviceShares() {
+  const keysToRemove: string[] = [];
+  await iterate((value, key, iterationNumber) => {
+    if (key.startsWith(`${DEVICE_SHARE_KEY}_`)) {
+      keysToRemove.push(key);
+    }
+  });
+  for (const key of keysToRemove) {
+    // eslint-disable-next-line no-await-in-loop
+    await removeItem(key);
+  }
+}
+
+function arrayBufferToBase64(buffer: ArrayBuffer) {
+  let binary = '';
+  const bytes = new Uint8Array(buffer);
+  const len = bytes.byteLength;
+  for (let i = 0; i < len; i++) {
+    binary += String.fromCharCode(bytes[i]);
+  }
+  return window.btoa(binary);
+}
+
+export function base64ToArrayBuffer(base64: string) {
+  const binaryString = window.atob(base64);
```

Comment:
I suggest this change for mobile compatibility

```suggestion
const binaryString = strToUrlBase64(base64);
```